Added a how-to for deploying the custom domains operator from a developer branch for testing (#78)
openshift/custom-domains-operator: DEPLOYING-FOR-TESTING.md

# Deploying the operator for testing

This section describess testing your local changes of the custom domains operator. 

> First, create a cluster in stage where you plan to deploy the custom domains operator.

Reviewer: To make less RH-specific:
```suggestion
First, create an OSD cluster in your staging environment.
```

### Building and pushing image in Quay

> To deploy the operator from your branch, you have to create an image in Quay. First, log in to quay and create a folder where you will push the image. In this example I will build the image in quay and push it to my quay account **aliceh** to a folder named **cdo**  with a tag **0**.

Reviewer: ```suggestion
To deploy the operator from your branch, you have to create an image in Quay under your personal account. For example, this will build the image in quay and push it to my personalquay account **aliceh** to a repository named **cdo**  with a tag **0**.
```

```

> docker build . -f build/Dockerfile -t quay.io/ahubenko/cdo:0

Reviewer: ```suggestion
docker build . -f build/Dockerfile -t quay.io/foobar/cdo:0
```

```

```

> docker push quay.io/aliceh/cdo:0 

Reviewer: ```suggestion
docker push quay.io/foobar/cdo:0 
```

> ```

Reviewer 2: I think we have Makefile rules to build and push images (`docker-build` and `docker-push` iirc). Is there any advantage or reason to directly invoke `docker`/`podman` here?

Author: Seems like this would require setting env vars which is not obvious.
Would it be all right to add this alternate way to build/push later?

### Pause the sync sets for your cluster

> To pause sync sets for your cluster, log in to hive shard and run this command.

Reviewer: ```suggestion
To pause sync sets for your cluster, log into the hive shard that manages your cluster and run this command.
```

          - preference:
              matchExpressions:
              - key: node-role.kubernetes.io/infra
                operator: Exists
            weight: 1
      tolerations:
        - effect: NoSchedule
          key: node-role.kubernetes.io/infra
          operator: Exists
      serviceAccountName: custom-domains-operator
      containers:
        - name: custom-domains-operator
          image: REPLACE_ME/custom-domains-operator:latest
          command:
          - custom-domains-operator
          args:
            - "--zap-log-level=debug"
            - "--zap-encoder=console"
          imagePullPolicy: Always
          env:
            - name: WATCH_NAMESPACE
              value: ""
            - name: POD_NAME
              valueFrom:
                fieldRef:
                  fieldPath: metadata.name
            - name: OPERATOR_NAME
              value: "custom-domains-operator"
```
replace the line 

```
image: REPLACE_ME/custom-domains-operator:latest
```
with the actual location of the image in quay, for example

```
image: quay.io/aliceh/cdo:0
```
Finally, deploy the contentes of the deploy directory:
```
oc apply -f deploy/
```

> 

Reviewer: Remove extra whitespace please.

> 

Reviewer: You can also use `oc set image deployment/custom-domains-operator quay.io/foobar/cdo:<new-tag>` for follow on image updates.

Author: Thanks!

        name: custom-domains-operator
    spec:
      affinity:
        nodeAffinity:
          preferredDuringSchedulingIgnoredDuringExecution:
          - preference:
              matchExpressions:
              - key: node-role.kubernetes.io/infra
                operator: Exists
            weight: 1
      tolerations:
        - effect: NoSchedule
          key: node-role.kubernetes.io/infra
          operator: Exists
      serviceAccountName: custom-domains-operator
      containers:
        - name: custom-domains-operator
          image: REPLACE_ME/custom-domains-operator:latest
          command:
          - custom-domains-operator
          args:
            - "--zap-log-level=debug"
            - "--zap-encoder=console"
          imagePullPolicy: Always
          env:
            - name: WATCH_NAMESPACE
              value: ""
            - name: POD_NAME
              valueFrom:
                fieldRef:
                  fieldPath: metadata.name
            - name: OPERATOR_NAME
              value: "custom-domains-operator"
```
replace the line 

```
image: REPLACE_ME/custom-domains-operator:latest
```
with the actual location of the image in quay, for example

```

> image: quay.io/aliceh/cdo:0

Reviewer: ```suggestion
image: quay.io/foobar/cdo:0
```
